tensorflow/core/kernels/data/range_dataset_op_test.cc

```diff
@@ -30,194 +32,75 @@ namespace tensorflow {
-  Status MakeOpDef() {
-    DataType value_type = tensorflow::DataTypeToEnum<T>::value;
-    std::vector<PartialTensorShape> shapes({{}});
-    DataTypeVector dtypes({value_type});
-
-    TF_RETURN_IF_ERROR(NodeDefBuilder(kNodeName, kOpName)
-                           .Input(FakeInput(DT_INT64))
-                           .Input(FakeInput(DT_INT64))
-                           .Input(FakeInput(DT_INT64))
-                           .Attr("output_types", dtypes)
-                           .Attr("output_shapes", shapes)
-                           .Finalize(node_def()));
-    TF_RETURN_IF_ERROR(InitOp());
-    return Status::OK();
-  }
-
-  Status MakeDataset(int64 start, int64 end, int64 step, int output_index,
-                     int thread_num, int cpu_num) {
-    // Clear inputs_ to enable the repeated update of dataset inputs.
+  DatasetBase* CreateRangeDataset(int64 start, int64 end, int64 step) {
     inputs_.clear();
-    AddInputFromArray<int64>(TensorShape({}), {start});
-    AddInputFromArray<int64>(TensorShape({}), {end});
-    AddInputFromArray<int64>(TensorShape({}), {step});
-
-    TF_RETURN_IF_ERROR(InitThreadPool(thread_num));
-    TF_RETURN_IF_ERROR(InitFunctionLibraryRuntime({}, cpu_num));
-    TF_RETURN_IF_ERROR(RunOpKernel());
-    TF_RETURN_IF_ERROR(InitDatasetFromContext(output_index));
-    return Status::OK();
-  }
-
-  Status MakeIterator() {
-    iterator_context_.reset();
-    iterator_context_ = MakeUnique<IteratorContext>(context_.get());
-    iterator_.reset();
-    return dataset_->MakeIterator(iterator_context_.get(), "Iterator",
-                                  &iterator_);
+    TF_CHECK_OK(CreateRangeDatasetOpKernel<T>("range", &kernel_));
+    TF_CHECK_OK(AddDatasetInputFromArray<int64>(
+        &inputs_, kernel_->input_types(), TensorShape({}), {start}));
+    TF_CHECK_OK(AddDatasetInputFromArray<int64>(
+        &inputs_, kernel_->input_types(), TensorShape({}), {end}));
+    TF_CHECK_OK(AddDatasetInputFromArray<int64>(
+        &inputs_, kernel_->input_types(), TensorShape({}), {step}));
+
+    TF_CHECK_OK(CreateOpKernelContext(kernel_.get(), &inputs_, &context_));
+    TF_CHECK_OK(CheckOpKernelInput(*kernel_, inputs_));
+    DatasetBase* dataset;
+    TF_CHECK_OK(CreateDataset(kernel_.get(), context_.get(), &dataset));
+    return dataset;
   }
 
-  Status GetNext(std::vector<Tensor>* out_tensors, bool* end_of_sequence) {
-    return iterator_->GetNext(iterator_context_.get(), out_tensors,
-                              end_of_sequence);
-  }
-
-  const string& name() const { return dataset_->name(); }
-
  protected:
   DatasetBase* dataset_;
-  FunctionLibraryRuntime* flr_;  // not owned
-  std::function<void(std::function<void()>)> runner_;
-  std::unique_ptr<DeviceMgr> device_mgr_;
-  std::unique_ptr<FunctionLibraryDefinition> lib_def_;
-  std::unique_ptr<ProcessFunctionLibraryRuntime> pflr_;
-  std::unique_ptr<thread::ThreadPool> thread_pool_;
+  std::unique_ptr<OpKernel> kernel_;
```

Comment:
do we need any of this state to be member variables or could it just be local variables?